tests/test-notice-visible-correct-role.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * TestNoticeVisibleForCorrectRole Testcase
+ *
+ * @package Astra_Notices
+ */
+
+/**
+ * Tests to verify the notices are visible to correct users.
+ */
+class TestNoticeVisibleForCorrectRole extends WP_UnitTestCase {
+
+	/**
+	 * Setup.
+	 *
+	 * @return void
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->editor_user_id = self::factory()->user->create(
+			array(
+				'role' => 'editor',
+			)
+		);
+
+		$this->subscriber_user_id = self::factory()->user->create(
+			array(
+				'role' => 'subscriber',
+			)
+		);
+
+		$this->administrator_user_id = self::factory()->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	/**
+	 * Test to verify the notice is visible to user's with given user level/capability.
+	 */
+	public function test_user_can_see_notice() {
+		$astra_notices = new Astra_Notices();
```

Comment:
I just remembered this - This was done to overcome the singleton affecting the tests.

If we use singleton, a notice registered from one unit test is also available in the next test as the same instance is carried forward in those tests. By initializing the class separately I am resetting the registered notices so that each test runs separately.

More thoughts on this here - https://github.com/brainstormforce/astra-notices/pull/1#discussion_r665052672

Reply:
@sandeshjangam https://github.com/brainstormforce/astra-notices/pull/13/ Can you check this?